### ISSUE TITLE
getJSDocComment doesn't work with a ClassExpression wrapped by the ExportNamedDeclaration

### ISSUE THREAD

**Issue by berlysia** [eslint/eslint#10983]
**Tell us about your environment**

* **ESLint Version:** 5.7.0
* **Node Version:** 10.12.0
* **npm Version:** 6.4.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{ rules: { checker: "error" }, parserOptions: { ecmaVersion: 6, sourceType: "module" } }
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const assert = require("assert");
const sinon = require("sinon");
const Linter = require("eslint/lib/linter");
const linter = new Linter();

const code = [
  "/** Desc*/",
  "export const Hoge = class Foo {};"
].join("\n");

function assertJSDoc(node) {
  const sourceCode = linter.getSourceCode();
  const jsdoc = sourceCode.getJSDocComment(node);

  assert(jsdoc); // fails in here
  assert.strictEqual(jsdoc.type, "Block");
  assert.strictEqual(jsdoc.value, "* Desc");
}

const spy = sinon.spy(assertJSDoc);
linter.defineRule("checker", () => ({ "ClassExpression": spy }));
linter.verify(code, { rules: { checker: "error" }, parserOptions: { ecmaVersion: 6, sourceType: "module" } });
assert(spy.calledOnce, "Event handler should be called.");
```

`npm i eslint sinon && node code_in_described_above.js` 

**What did you expect to happen?**
process exits successfully

**What actually happened? Please include the actual, raw output from ESLint.**
at `assert(jsdoc); // fails in here`, `jsdoc` is null

**Comment by kaicataldo** [comment]
Does this also occur with `export class Foo {};`? Or is limited to an exporting an assignment?

**Comment by berlysia** [comment]
It's limited to this case.

<details>
<summary>
I've already check some patterns listed below:

```js
export const Foo = class Bar {}; // only this case fails the test
class Foo {}
const Foo = class Bar {};
Foo = class Bar {};
export class Foo {}
export default class Foo {}
```
</summary>
output:

```
jsdoc is null /** Desc*/
export const Foo = class Bar {};
```

check code:
```js
const assert = require("assert");
const sinon = require("sinon");
const Linter = require("eslint/lib/linter");
const linter = new Linter();

const codes = [
  "/** Desc*/\nexport const Foo = class Bar {};", // only this pattern
  "/** Desc*/\nclass Foo {}", // ClassDeclaration
  "/** Desc*/\nconst Foo = class Bar {};", // ClassExpression with VariableDeclaration
  "/** Desc*/\nFoo = class Bar {};", // ClassExpression with AssignmentExpression
  "/** Desc*/\nexport class Foo {}", // named exported ClassDeclaration
  "/** Desc*/\nexport default class Foo {}" // default exported ClassDeclaration
];

function assertJSDoc(node) {
  const sourceCode = linter.getSourceCode();
  const jsdoc = sourceCode.getJSDocComment(node);

  assert(jsdoc, "jsdoc is null");
  assert.strictEqual(jsdoc.type, "Block");
  assert.strictEqual(jsdoc.value, "* Desc");
}

codes.forEach(code => {
  try {
    const spy = sinon.spy(assertJSDoc);
    linter.defineRule("checker", () => ({
      ClassExpression: spy,
      ClassDeclaration: spy
    }));
    linter.verify(code, {
      rules: { checker: "error" },
      parserOptions: { ecmaVersion: 6, sourceType: "module" }
    });
    assert(spy.calledOnce, "Event handler should be called.");
  } catch (e) {
    console.error(e.message, code);
  }
});

```
</details>

**Comment by nzakas** [comment]
Thanks for the suggestion. As of yesterday, we have decided to officially [end-of-life JSDoc support](https://eslint.org/blog/2018/11/jsdoc-end-of-life) in ESLint. All JSDoc features are now deprecated and we won't be fixing bugs or making any further improvements to those features.

Thanks for understanding and we apologize for any inconvenience.
